edb/common/parsing.py

```diff
@@ -147,36 +148,31 @@ def __init_subclass__(cls, *, is_internal=False, **kwargs):
                 setattr(cls, name, a)
 
 
-class ListNonterm(Nonterm, is_internal=True):
-    def __init_subclass__(cls, *, element, separator=None, is_internal=False,
-                          **kwargs):
-
-        if not is_internal:
-            element_name = ListNonterm._component_name(element)
-            separator_name = ListNonterm._component_name(separator)
-
-            if separator_name:
-                tail_prod = (
-                    lambda self, lst, sep, el: self._reduce_list(lst, el)
-                )
-                tail_prod_name = 'reduce_{}_{}_{}'.format(
-                    cls.__name__, separator_name, element_name)
-            else:
-                tail_prod = (
-                    lambda self, lst, el: self._reduce_list(lst, el)
-                )
-                tail_prod_name = 'reduce_{}_{}'.format(
-                    cls.__name__, element_name)
-            setattr(cls, tail_prod_name, tail_prod)
-
-            setattr(cls, 'reduce_' + element_name,
-                lambda self, el: self._reduce_el(el))
+class ListHelper(Nonterm, is_internal=True):
```

Comment:
Does it help any to have this in a separate class? These could probably be staticmethods on `ListNonterm` itself, I think